### PR TITLE
Github action to deploy `docs` folder to Github Pages

### LINKED ISSUE #9: Deploy documentation as readable site
Implement mechanism to deploy the `docs` (or similar) folder in the repo to be deployed as a readable navigable documentation site.

### PR DESCRIPTION
**Pull request by shuaibq3**
Fixes #9 

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Takes the markdown files in the `docs` folder and deploys them to github pages in the link https://craftsmenltd.github.io/BloodConnect .
ar description of the problem or feature being addressed.

## How to test
The current implementation adds a github action that triggers when a commit is pushed to the current branch (issue-9-doc-as-readable-site). If the action needs to be triggered with any other branch, the branch needs to be added to the environment from settings (https://github.com/withastro/docs/issues/1376#issuecomment-1226959256) to allow it to deploy to  `github-pages` environment. Also, the branch will need to be added to [action](https://github.com/CraftsmenLtd/BloodConnect/compare/issue-9-doc-as-readable-site?expand=1#diff-87e860af4b9db6c503ed680b6edb6333c6f8d7659f7d1a779a2487466bbc50f6R5) to trigger the workflow.

- Add markdown file to the `/docs`. Create a new commit and push to this branch. This should trigger the github action, which should deploy the site in the mentioned link. The link should have the new markdown file in the appropriate url.
- Add **non markdown** file to the `/docs`. Create a new commit and push to this branch. This should trigger the github action and deploy the site successfully. However, there should not be any changes in the site.

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [x] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.

## Reviewer Request (optional)

Please make sure to revert the commits but resetting HEAD and `force-push` to this branch to remove any unnecessary commits and files from his PR.

## Demo Link (optional)

https://craftsmenltd.github.io/BloodConnect/

## Additional Notes (optional)

Once this PR is approved, we will need to remove the additional [branch name](https://github.com/CraftsmenLtd/BloodConnect/compare/issue-9-doc-as-readable-site?expand=1#diff-87e860af4b9db6c503ed680b6edb6333c6f8d7659f7d1a779a2487466bbc50f6R8) from the action yml file and create another commit. Another approval for after pushing that commit may be needed to finally merge it to master.

Alternatively, the reviewer can push a followup commit with branch name removed and then approve the PR, with that commit added.

Removal of this branch name (and any additional branch names added) from the github-pages environment is needed after this PR is approved.

## Deploy risk & follow up

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: .github/workflows/deploy-docs.yml
Comment (SayefReyadh) [inline .github/workflows/deploy-docs.yml:8]: `issue-9-doc-as-readable-site` is this a default branch?

**Review by SayefReyadh — APPROVED** [review]
Consider resolving the feedbacks before merge.

File: .github/workflows/deploy-docs.yml
Comment (shuaibq3) [reply .github/workflows/deploy-docs.yml:8]: `issue-9-doc-as-readable-site` this is not a default branch. this has been added for testing purpose, and as mentioned in the PR description, needs to be removed after approval.
```
Once this PR is approved, we will need to remove the additional [branch name](https://github.com/CraftsmenLtd/BloodConnect/compare/issue-9-doc-as-readable-site?expand=1#diff-87e860af4b9db6c503ed680b6edb6333c6f8d7659f7d1a779a2487466bbc50f6R8) from the action yml file and create another commit. Another approval for after pushing that commit may be needed to finally merge it to master.
```

As mentioned, will remove this with a follow up commit and merge

**Pull request merged**
